src/main/scala/beam/physsim/jdeqsim/JDEQSimRunner.scala

```diff
@@ -107,23 +108,25 @@ class JDEQSimRunner(
       }
       maybeCaccSettings.foreach(_.roadCapacityAdjustmentFunction.reset())
 
-      legHistogram.getLegModes.forEach(mode => {
-        new PlotGraph().writeGraphic(
-          legHistogram,
-          controlerIO,
-          s"$currentPhysSimIter.physsimTripHistogram",
-          "time (binSize=<?> sec)",
-          mode,
-          agentSimIterationNumber,
-          beamConfig.beam.outputs.stats.binSize
-        )
-      })
-      linkStatsGraph.notifyIterationEnds(agentSimIterationNumber, travelTimeCalculator.getLinkTravelTimes);
-      eventToHourFrequency.notifyIterationEnds(
-        new IterationEndsEvent(beamServices.matsimServices, agentSimIterationNumber)
-      );
-      linkStatsGraph.notifyIterationEnds(agentSimIterationNumber, travelTimeCalculator.getLinkTravelTimes)
-      physsimSpeedHandler.notifyIterationEnds(agentSimIterationNumber)
+      parallelExecution(
+        legHistogram.getLegModes.forEach(mode => {
+          new PlotGraph().writeGraphic(
+            legHistogram,
+            controlerIO,
+            s"$currentPhysSimIter.physsimTripHistogram",
+            "time (binSize=<?> sec)",
+            mode,
+            agentSimIterationNumber,
+            beamConfig.beam.outputs.stats.binSize
+          )
+        }),
+        linkStatsGraph.notifyIterationEnds(agentSimIterationNumber, travelTimeCalculator.getLinkTravelTimes),
+        eventToHourFrequency.notifyIterationEnds(
+          new IterationEndsEvent(beamServices.matsimServices, agentSimIterationNumber)
+        ),
+        linkStatsGraph.notifyIterationEnds(agentSimIterationNumber, travelTimeCalculator.getLinkTravelTimes),
```

Comment:
Probably this is a bug introduced before. The same line is used twice:
```
linkStatsGraph.notifyIterationEnds(agentSimIterationNumber, travelTimeCalculator.getLinkTravelTimes)
```

Reply:
I think you are right.